garage/envs/base.py

```diff
@@ -1,17 +1,121 @@
+"""Wrapper class that converts gym.Env into GarageEnv."""
+from abc import ABC, abstractmethod
 import collections
+import warnings
 
-from cached_property import cached_property
+import gym
 
-from garage.envs import EnvSpec
+from garage.core import Parameterized
+from garage.core import Serializable
 
-_Step = collections.namedtuple("Step",
-                               ["observation", "reward", "done", "info"])
 
+class GarageEnv(ABC, gym.Wrapper, Parameterized, Serializable):
```

Comment:
don't inherit from ABC first--that means super() will call the ABC constructor.

most people following this pattern use the metaclass param instead to avoid this problem

```python
from abc import ABCMeta

class MyABC(metaclass=ABCMeta):
    pass
```

Reply:
```python
class GarageEnv(gym.Wrapper, Parameterized, Serializable, metaclass=ABCMeta):
 ...